src/Compiler/CompilerOptions.cpp

```diff
@@ -0,0 +1,257 @@
+    llvm::cl::desc(
+        "invoke extra onnx transform pass(shape inference, constant and etc.)"),
+    llvm::cl::init(0), llvm::cl::cat(OnnxMlirOptions));
+
+llvm::cl::opt<string> shapeInformation("shapeInformation",
+    llvm::cl::desc(
+        "Custom shapes for the inputs of the ONNX model, e.g. setting static "
+        "shapes for dynamic inputs.\n"
+        "\"value\" is in the format of "
+        "\"INPUT_ID1:D1xD2x...xDn,INPUT_ID2:D1xD2x...xDn, ...\",\n"
+        "where \"INPUT_ID1, INPUT_ID2, ...\" are input indices starting from "
+        "0, and\n"
+        "\"D1, D2, ...\" are dimension sizes (positive integers of -1 for "
+        "unknown dimensions)"),
+    llvm::cl::value_desc("value"), llvm::cl::cat(OnnxMlirOptions));
+
+llvm::cl::opt<std::string> mtriple("mtriple",
+    llvm::cl::desc("Override target triple for module"),
+    llvm::cl::value_desc("LLVM target triple"), llvm::cl::cat(OnnxMlirOptions),
+    llvm::cl::ValueRequired);
+
+llvm::cl::opt<std::string> mcpu("mcpu", llvm::cl::desc("Target cpu"),
+    llvm::cl::value_desc("Target a specific CPU type"),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::ValueRequired);
+
+llvm::cl::opt<std::string> march("march",
+    llvm::cl::desc("Target architecture to generate code for"),
+    llvm::cl::value_desc("Target a specific architecture type"),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::ValueRequired);
+
+llvm::cl::opt<bool> VerboseOutput("v", llvm::cl::desc("Use verbose output"),
+    llvm::cl::init(false), llvm::cl::cat(OnnxMlirOptions));
+
+llvm::cl::opt<std::string> Xopt("Xopt",
+    llvm::cl::desc("Arguments to forward to LLVM's 'opt' option processing"),
+    llvm::cl::value_desc("A valid LLVM's 'opt' option"),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::Hidden, llvm::cl::ValueRequired);
+
+llvm::cl::opt<std::string> Xllc("Xllc",
+    llvm::cl::desc("Arguments to forward to LLVM's 'llc' option processing"),
+    llvm::cl::value_desc("A valid LLVM's 'llc' option"),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::Hidden, llvm::cl::ValueRequired);
+
+llvm::cl::opt<std::string> mllvm("mllvm",
+    llvm::cl::desc(
+        "Arguments to forward to LLVM's 'opt' and 'llc' option processing"),
+    llvm::cl::value_desc("A valid LLVM's 'opt' and 'llc' option"),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::Hidden, llvm::cl::ValueRequired);
+
+llvm::cl::opt<OptLevel> OptimizationLevel(
+    llvm::cl::desc("Optimization levels:"),
+    llvm::cl::values(clEnumVal(O0, "Optimization level 0 (default)."),
+        clEnumVal(O1, "Optimization level 1."),
+        clEnumVal(O2, "Optimization level 2."),
+        clEnumVal(O3, "Optimization level 3.")),
+    llvm::cl::init(O0), llvm::cl::cat(OnnxMlirOptions));
+} // namespace onnx_mlir
+
+// =============================================================================
+// Methods for setting and getting compiler variables.
+
+// Triple.
+void setTargetTriple(const std::string &triple) {
```

Comment:
Would be nice to place all these functions inside the `onnx_mlir` namespace. As they are they invade the user namespace.

Reply:
Done. They are inside the `onnx_mlir` namespace now.